### ISSUE TITLE
Ref<T> still does not work.

### ISSUE THREAD

**Issue by norton-corbett** [godotengine/godot-cpp#462]
Godot: 3.2.3 stable
OS: Windows 10

Not sure if anybody still cares about this. I left a comment on https://github.com/godotengine/godot-cpp/issues/417 back at the beginning of August and there's been no response.

This crashes on Windows:

```c++
#pragma once

#include <Godot.hpp>
#include <Node.hpp>

using namespace godot;

class Thing : public Reference
{
	GODOT_CLASS(Thing, Reference);

public:

	static void _register_methods()
	{
		register_method("_init", &Thing::_init);
	}

	void _init() {}
};

class ThingMaker : public Node
{
	GODOT_CLASS(ThingMaker, Node);

public:

	static void _register_methods()
	{
		register_method("_init", &ThingMaker::_init);
		register_method("make_thing", &ThingMaker::make_thing);
	}

	void _init() {}

	Ref<Thing> make_thing()
	{
		Thing* new_thing = Thing::_new();

		Ref<Thing> ref = Ref<Thing>(new_thing); // crash

		return ref;
	}
};
```

Using the old `__internal_constructor()` trick also crashes.

I would debug it myself but the code for this stuff is so nebulous I honestly have no idea where to start.

**Comment by beroso** [comment]
I tested your code and it's working fine.
I only experienced the crash when trying to enforce static typing on GDScript side.

This works:
```gdscript
var tm := ThingMaker.new()
var thing = tm.make_thing()
```

And this crashes:
```gdscript
var tm := ThingMaker.new()
var thing : Thing = tm.make_thing() # crash!
```

It fails whenever I tried to cast `make_thing()`'s return. Another example:

```gdscript
func print_thing(thing):
	print(thing)
	

func print_thing2(thing : Thing):
	print(thing)


func _ready():
	var tm := ThingMaker.new()
	var t = tm.make_thing()
	print_thing(t) # prints [Reference:###]
	print_thing2(t) # prints Null
	#var t2 = (t as Thing) # crash!
```

**Comment by norton-corbett** [comment]
For me it consistently crashes in C++ before returning from `make_thing()`, at the line I indicated, regardless of what I am doing in GDScript. I think these are two separate bugs and I guess it's environment dependent somehow.

**Comment by norton-corbett** [comment]
Ok I managed to fix it and it was obviously user error but I'm not sure what. Fetching latest changes from the repo and rebuilding did not fix it. Doing a clean and rebuild of everything did not fix it. What actually fixed it was deleting the entire godot-cpp folder and re-cloning from github.
I checked and I didn't have any local changes to the source so maybe something got messed up in the generated bindings, No idea.
